gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+// Make a change
+ChangeRequest.Builder changeBuilder = ChangeRequest.builder().add(toCreate);
+// Build and apply the change request to our zone
+ChangeRequest changeRequest = zone.applyChangeRequest(changeBuilder.build());
+```
+
+The `addRecord` (and also `records`) function of `DnsRecord.Builder` accepts records in the form of
+strings (and a list of strings respectively). The format of the strings (and the number of strings
+in the list respectively) depends on the type of the DNS record to be added. More information on the
+supported DNS record types and record formats can be found [here](https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types).
+
+When the change request is applied, it is registered with the Cloud DNS service for processing. We
+can wait for its completion as follows:
+
+```java
+while (ChangeRequest.Status.PENDING.equals(changeRequest.status())) {
+  Thread.sleep(500L);
+  changeRequest = dns.getChangeRequest(zone.name(), changeRequest.id());
+}
+System.out.println("The change request has been applied.");
+```
+
+Change requests are applied atomically to all the assigned DNS servers at once. Note that when this
+happens, it may still take a while for the change to be registered by the DNS cache resolvers in the
+Internet. See more on this topic [here](https://cloud.google.com/dns/monitoring).
+
+#### Listing Zones and DNS Records
+Suppose that you've added more zones and DNS records in them, and now you want to see what they are.
+First, import the following class for easy iteration:
+
+```java
+import java.util.Iterator;
+```
+
+Then add the following code to list all your zones and DNS records.
+
+```java
+// List all your zones
+Iterator<Zone> projectZones = dns.listZones().iterateAll();
+System.out.println("My zones:");
+while (projectZones.hasNext()) {
+  System.out.println(projectZones.next());
+}
+
+// List the DNS records in a particular zone
+Iterator<DnsRecord> recordIterator = zone.list().iterateAll();
+System.out.println(String.format("DNS records inside %s:", zone.name());
+while (recordIterator.hasNext()) {
+  System.out.println(recordIterator.next());
+}
+```
+
+You can also list the history of change requests that were applied to a zone:
+
+```java
+// List the change requests applied to a particular zone
+Iterator<ChangeRequest> changeIterator = zone.listChangeRequests().iterateAll();
+System.out.println(String.format("The history of changes in %s:", zone.name());
+while (changeIterator.hasNext()) {
+  System.out.println(changeIterator.next());
+}
+```
+
+#### Deleting Zones
+
+If you no longer want to host a zone in Cloud DNS, you can delete it. However, you need to empty it
+first by creating a change that deletes all such records. The only thing you should be careful about
+is not to delete the default SOA and NS records for the zone.
+
+```java
+// prepare change for deleting all records
+List<DnsRecord.Type> defaultTypes = ImmutableList.of(DnsRecord.Type.SOA, DnsRecord.Type.NS);
```

Comment:
Can be done. I am not sure if building a more complicated and composed operations is our goal here though. One could come up with a few more that would be handy, such as waiting for a change to complete or a test for emptiness for a zone.